src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1342,6 +1278,27 @@ static bool processContextAttributeVector (ContextElement*
 
     }
 
+    /* Special processing in the case of DELETE, to avoid "deleting too much" in the case of using
+     * metadata ID. If metadata ID feature were removed this ugly piece of code wouldn't be needed ;)
+     * FIXME P3: note that in the case of Active/Active configuration it could happen that
+     * the attrsName get desyncrhonized, e.g. starting situation is A-ID1 and A-ID2, node
+     * #1 process DELETE A-ID1 and node #2 process DELETE A-ID2, each one thinking than the
+     * other A-IDx copy will be there at the end, thus nobody includes it in the toPull array
+     */
```

Comment:
I understand that this limitation is acceptable, given that DELETE operations on attributes with ID will not be so frequent in an Active/Active scenario. @mrutid what do you think?

Any way, the metadata ID stuff creates *a lot* of complexity in the code for a rather useless feature (see http://stackoverflow.com/questions/29699180/can-a-orion-context-broker-entity-have-two-attributes-with-the-same-name-but-dif). I will open an internal discussion about deprecating it.